talk.md

```diff
@@ -0,0 +1,487 @@
+      it, but perhaps I’ve just missed it. Come at me, at the bar, and set me straight!
+
+## Errors vs errors
+
+First of all, I wanted to take a step back and talk about errors in general. The nomenclature around these can get
+confusing, suffice to say that during this session we’ll talk about these two types:
+
+1. Errors that occurred during query execution that were unexpected and _could_ lead to corrupted data. We’ll refer to
+   these as (top-level) ‘GraphQL errors’, going forward.
+
+   These could be due to hardware failures, such as running out of memory or disk space, network failures, or unexpected
+   upstream data etc.
+
+   When these occur, `graphql-js` will return `null` for the field that triggered the error and serialize the error into
+   the top-level `errors` list, next to the successful response `data`. (Presumably other implementations
+   follow this reference implementation.)
+
+   ```json
+   {
+     data: { artwork: { artist: { name: "Vincent van Gogh", leftEarSize: null } } },
+     errors: [
+       {
+         message: "An unexpected error occurred",
+         path: ["artwork", "artist", "leftEarSize"]
+       }
+     ]
+   }
+   ```
+
+1. Exceptions to these are errors that are known to occur and are expected to be handled by the user of an API. We’ll
+   refer to these as ‘exceptions’, going forward.
+
+   By default these are treated equally by `graphql-js` to top-level GraphQL errors, if uncaught.
+
+We will **not** be speaking about errors that occur _outside_ of query execution, such as network failures reaching the
+GraphQL server, parsing a syntactically incorrect document, or passing variables that don’t satisfy the type-system; as
+these will all lead to a query being rejected wholesale and are solve-able using traditional means, such as a `4xx` HTTP
+status code or `5xx` in some cases.
+
+## What is the problem we’re trying to solve?
+
+Because with GraphQL we’re usually requesting data for multiple resources, there may be a situation where some fields
+resolve successfully and some may fail. This is also why, when using an HTTP transport layer, the advice is to always
+respond with a HTTP 200 (ok) status. Determining how to process the response is left up to the client.
+
+So how _do_ we model errors in such a way that they can be meaningful and in context of their origin?
+
+* What if you want to render partial data?
+
+  - Maybe the failed data is unrelated to other components that you were also requesting data for.
+
+    [TODO: Insert screenshot of page that has a Recently Viewed artwork rail]
+
+  - Or the data that failed was part of a list and other entries can still be rendered just fine.
+
+    [TODO: Insert screenshot of artwork grid]
+
+* Or what if you’d (additionally) like to communicate the error in your interface?
+
+  - Perhaps you’d want to display that some functionality only works when authorized.
+
+    [TODO: Come up with example on artsy and insert screenshot]
```

Comment:
Definitely having a hard time with this one. It doesn’t need to be around authorization at all, btw.